common/sched.c

```diff
@@ -94,16 +95,17 @@ static task_t *create_task(void) {
     return task;
 }
 
-static void prepare_task(task_t *task, const char *name, task_func_t func, void *arg) {
-    if (!task)
-        return;
+static int prepare_task(task_t *task, const char *name, task_func_t func, void *arg) {
+    if (!task || get_task_by_name(name))
```

Comment:
I would drop the check here and only check in `new_task`

Reply:
What if you call `prepare_task` directly? Or are you assuming someone won't in the future because it is `static`?

Reply:
Yes, for now it is `static` and I am assuming only `new_task` uses it. Alternatively we could keep the check in `prepare_task` and remove the one from `new_task` and simply add `destroy_task()` to cleanup when `prepare_task` fails after `create_task` has been already called.